spring-integration-core/src/main/java/org/springframework/integration/gateway/GatewayProxyFactoryBean.java

```diff
@@ -1020,6 +1032,11 @@ protected void doStop() {
 		}
 	}
 
+	@Override
+	public void destroy() {
+		this.gatewayMap.values().forEach(gw -> gw.destroy());
```

Comment:
```suggestion
		this.gatewayMap.values().forEach(MethodInvocationGateway::destroy);
```

How do we live without this before?
Should we consider to back-port it as a bug?
What is a consequence not having this `destroy()`?